### PR TITLE
Ecommerce site - non skeleton changes

### PR DESCRIPTION
**Pull request by yuriihohan**
# **My Changes on StoreFront page + Admin Part**

Create a single page web app that represents Ecommerce frontpage. Header and footer are static. Create a simple admin panel (http://guides.rubyonrails.org/engines.html), that lets you edit the content on the front page in sections as follows:

- Slider image, link and button text
- Featured products - title, price and image
- Inspiration images text and links
- Shipping block titles and text

 
You can use this [html](https://drive.google.com/file/d/1ByuKJFo9dnHrLMyN-QHPgVtjNO7YeEiw/view) template for this task. You can leave static files like css/js in html.

### CODE REVIEW

File: rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+$(document).ready(function(){
+    is_newForm = $('form').attr('id') === 'new_inspiration_block';
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js]: Same code as in feature_products.js

File: rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/slider_images.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
+$(document).ready(function () {
+    is_newForm = $('form').attr('id') === 'new_inspiration_block';
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/slider_images.js]: Same code as in feature_products.js

File: rails_exercises/ecommerce_front/admin/app/controllers/admin/feature_products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        render :new
+      end
+    end
+
+    def edit; end
+
+    def update
+      @feature_product.update_attributes(feature_product_params)
+
+      if @feature_product.save
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/controllers/admin/feature_products_controller.rb]: `update_attributes` already saves the record.
If you want to save separately you can use `assign_attributes`

File: rails_exercises/ecommerce_front/admin/app/controllers/admin/feature_products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    end
+
+    private
+
+    def set_feature_product
+      @feature_product = FeatureProduct.find(params[:id])
+    end
+
+    def feature_product_params
+      params[:feature_product]
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/controllers/admin/feature_products_controller.rb]: Allowing whatever parameters that come from outside is considered insecure.
Rails 4 has [strong parameters](http://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters) for that.
In Rails 3 you can do poor man's strong parameters - `params[:feature_product].slice(:id, :name, ...<other explicitly allowed params>)`

File: rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js (hunk 1 above)
Comment (yuriihohan) [reply rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js]: Do you think moving it to **utils** file is a good enough solution?

File: rails_exercises/ecommerce_front/admin/app/views/admin/feature_products/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    <%= f.text_field :header, class: 'input-xxlarge' %>
+  </div>
+
+  <div class="field">
+    <%= f.label :price %>
+    <%= f.number_field :price,  step: 0.01, class: 'input-xxlarge' %>
+  </div>
+  <div class="field">
+    <%= f.file_field :image, class: 'new_upload'  if @feature_product.id.nil?%>
+    <%= f.file_field :image, class: 're_upload'  unless @feature_product.id.nil?%>
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/views/admin/feature_products/_form.html.erb]: `<%= f.file_field :image, class: (@feature_product.persisted? ? 'new_upload' : 're_upload') %>`

File: rails_exercises/ecommerce_front/admin/app/views/admin/feature_products/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<h1 class="form_title">Edit Feature Product</h1> 
+
+<div class="forms">
+ <%= render 'form', task: @feature_product %>
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/views/admin/feature_products/edit.html.erb]: Copy/paste from the TODO app exercise?

File: rails_exercises/ecommerce_front/admin/app/views/admin/shipping_info/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<h1>Configure Shipping Info Section</h1>
+
+<ul>
+  <% @shipping_info_blocks.each do |info_block| %>
+    <li>
+      <div>
+        <div class = "controls">
+          <span class = "ud-icons">
+            <%= link_to "<i class='far fa-edit'></i>".html_safe, shipping_info_edit_path(info_block) %>
```
Comment (vlebedeff) [inline rails_exercises/ecommerce_front/admin/app/views/admin/shipping_info/index.html.erb]: Avoid `html_safe` everywhere you can. Use `link_to` helper with a block

File: rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js (hunk 1 above)
Comment (vlebedeff) [reply rails_exercises/ecommerce_front/admin/app/assets/javascripts/admin/inspiration_blocks.js]: That's fine

**Review by vlebedeff — APPROVED** [review]

**Pull request merged**